v2/cli/init_command.go

```diff
@@ -0,0 +1,248 @@
+	} else if !dirExists {
+		fmt.Println("Creating .brigade directory...")
+		err = os.Mkdir(".brigade", 0755)
+		if err != nil {
+			return err
+		}
+	}
+
+	buf, err := execTemplate(newProjectFileBytes, fields)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade/project.yaml path exists.
+	// Confirm overwrite if so.
+	if err = checkFileOverwrite(
+		path.Join(".brigade", "project.yaml"),
+		"project.yaml",
+		buf,
+	); err != nil {
+		return err
+	}
+
+	// If git repo is provided, create a separate js/ts file
+	if fields.Git != "" {
+		gitBuf, gitErr :=
+			execTemplate(newGitBrigadeFileBytes, fields)
+		if gitErr != nil {
+			return gitErr
+		}
+
+		gitFileName := fmt.Sprintf("brigade.%s", fields.Language)
+
+		if err = checkFileOverwrite(
+			path.Join(".brigade", gitFileName),
+			gitFileName,
+			gitBuf,
+		); err != nil {
+			return err
+		}
+	}
+
+	// TODO: Right now the notes doesn't have template fields, add later!
+	notesBuf, err :=
+		execTemplate(newNotesFileBytes, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a NOTES.txt file
+	if err = checkFileOverwrite(
+		path.Join(".brigade", "NOTES.txt"),
+		"NOTES.txt",
+		notesBuf,
+	); err != nil {
+		return err
+	}
+
+	secretsBuf, err :=
+		execTemplate(newSecretsFileBytes, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a secrets.yaml file
+	if err = checkFileOverwrite(
+		path.Join(".brigade", "secrets.yaml"),
+		"secrets.yaml",
+		secretsBuf,
+	); err != nil {
+		return err
+	}
+
+	err = addToGitignore(".brigade/secrets.yaml")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Detect if .gitignore exists. If it does, add .brigade/secrets.yaml to it
+// If not, create it and add .brigade/secrets.yaml to it
+func addToGitignore(filepath string) error {
+	if fileExists, err := file.Exists("./.gitignore"); err != nil {
+		return err
+	} else if fileExists {
+		f, err :=
+			os.OpenFile(".gitignore", os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+		if err != nil {
+			return err
+		}
+
+		defer f.Close()
+
+		b, err := ioutil.ReadFile(".gitignore")
+		if err != nil {
+			return err
+		}
+		s := string(b)
+
+		// Only add to gitignore if it's not in the gitignore already
+		if !strings.Contains(s, filepath) {
+			if _, err = f.WriteString("\n" + filepath); err != nil {
+				return err
+			}
+		}
+	} else {
+		fmt.Println("Creating .gitignore...")
+		if err := ioutil.WriteFile(
+			".gitignore",
+			[]byte(filepath),
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func idMatch(id string) error {
+	idMatch, err := regexp.MatchString("^[a-z][a-z\\d-]*[a-z\\d]$", id)
+	if !idMatch || len(id) < 3 || len(id) > 63 {
+		return fmt.Errorf("invalid value %q for --id flag"+
+			" (3-63 characters, first char must be"+
+			" a letter, lowercase letters only)", id)
+	} else if err != nil {
+		return err
+	}
+	return nil
+}
+
+func normalizeLanguage(language string) (string, error) {
+	switch language {
+	case "ts", "typescript", "":
```

Comment:
I've suggested specifying the default [here](https://github.com/brigadecore/brigade/pull/1437/files#r647448934) so, I'd suggest _not_ accepting `""` here as being indicative of TS. If the default ever were to change (unlikely), we'd only want to change it in one place and the other place is probably the more obvious place where someone would look.